src/audio/host.c

```diff
@@ -131,6 +131,32 @@ static uint32_t host_dma_get_split(struct host_data *hd, uint32_t bytes)
 	return MAX(split_src, split_dst);
 }
 
+static int host_dma_set_config_and_copy(struct comp_dev *dev, uint32_t bytes)
+{
+	struct host_data *hd = comp_get_drvdata(dev);
+	struct dma_sg_elem *local_elem = hd->config.elem_array.elems;
+	int ret = 0;
+
+	local_elem->size = bytes;
+
+	/* reconfigure transfer */
+	ret = dma_set_config(hd->chan, &hd->config);
+	if (ret < 0) {
+		comp_err(dev, "host_dma_set_config_and_copy(): dma_set_config() failed, ret = %d",
+			 ret);
+		return ret;
+	}
+
+	ret = dma_copy(hd->chan, bytes, DMA_COPY_ONE_SHOT);
```

Comment:
this function is potentially called in a loop, so here you're firing several one-shot DMA copy operations in non-blocking mode (`DMA_COPY_BLOCKING` flag isn't set). I think this could be the reason for `-EBUSY` returned by Intel non-cAVS systems

Reply:
Thanks! This has been addressed and let's waiting for the CI to verify again. (looks like it's manually triggered)